app/main/views/users.py

```diff
@@ -19,14 +19,25 @@ def auth_user():
     json_payload = json_payload["authUsers"]
     validate_user_auth_json_or_400(json_payload)
 
-    user = User.query.filter(
-        User.email_address == json_payload['emailAddress'].lower()).first()
+    user = User.get_by_email_address(json_payload['emailAddress'].lower())
 
     if user is None:
         return jsonify(authorization=False), 404
-    elif encryption.checkpw(json_payload['password'], user.password):
+    elif encryption.authenticate_user(json_payload['password'], user):
+        user.logged_in_at = datetime.utcnow()
+        user.failed_login_count = 0
+        db.session.add(user)
+        db.session.commit()
+
         return jsonify(users=user.serialize()), 200
     else:
+        login_limit = current_app.config['DM_FAILED_LOGIN_LIMIT']
+        user.failed_login_count += 1
+        if user.failed_login_count >= login_limit:
+            user.locked = True
```

Comment:
We only export active and unlocked users from the Grails app, so no need to preserve either of these if we don't need them. Users are only locked for failed logins, no other reason.